# Programação Orientada a Objetos, Turma de 2016

## Segunda prática

*(Vale 5 pontos)*

---

### Triplas pitagóricas

Uma tripla pitagórica é uma tripla de números **inteiros** $(a, b, c)$ tal que:
$$
a > b > c > 0\\
a^2 = b^2 + c^2
$$

Por exemplo, $(5, 4, 3)$ é uma tripla pitagórica.

### Problema

Dado que existe uma e apenas uma tripla pitagórica $(a, b, c)$ para a qual $a + b + c = 1000$, escreva um código Python para encontrar e mostrar essa tripla.

Evite fazer mais cálculos do que o necessário. Use células de *markdown* para explicar o seu código. Lembre-se de testar se a resposta fornecida pelo seu programa satisfaz as condições (use código Python para fazer o teste em células apropriadas).

In [18]:
s= 1000

a, b, c= 0, 0, 0

for i in range(1, s):
    for j in range( 1, s-1):
        a=s-i-j
        if((a**2)== (i**2)+(j**2) and a>i>j):        
            b=i
            c=j
            print(a,b,c)
        
    
    

        

425 375 200


Atribui o valor 1000 a variavel s, que representa a soma dos termos a, b e c, e em seguida, atribui 0 as variaveis a, b e c, para inicializa-las. Em seguida usei um for para fazer com que i vá de 1 ao valor da soma, (i seria um auxiliar para o b, no caso) e outro for para que j vá de 1 a s-1 (s-1 pois j representa um auxiliar para c, e c tem que ser menor que b, no caso, o i). Dentro do segundo for, coloquei a expressão a=s-i-j, que dá o valor de a para cada i e j incrementado nos comandos for, respeitando a igualdade "a+b+c=1000". Tambem dentro do segundo for, coloquei um comando if para verificar se o a, o i e o j encontrados satisfazem a equação do teorema de pitágoras e a relação a>b>c>0. Quando o a, o i e o j satisfazerem as relações, b recebe o valor de i e c recebe o valor de j, e printa-se a,b,c. O resultado encontrado foi a=425, b=375 e c=200. 

---

* Você não precisa testar todas as combinações de `i` e `j`, pois há duplicações (por exemplo, (15, 12) e (12, 15)).
* Você não precisa percorrer `i` e `j` até `s`, pois $a>b>c$ e $a+b+c=1000$.
* Quando você acha uma solução, você imprime, mas continua procurando novas soluções. Isso não é necessário, pois o enunciado diz que há apenas uma solução.

-- *Gonzalo*